include/MixerProfiler.h

```diff
@@ -41,21 +42,34 @@ class MixerProfiler
 		m_periodTimer.reset();
 	}
 
-	void finishPeriod( sample_rate_t sampleRate, fpp_t framesPerPeriod );
+	void finishPeriod(sample_rate_t sampleRate, fpp_t framesPerPeriod);
 
 	int cpuLoad() const
 	{
 		return m_cpuLoad;
 	}
 
-	void setOutputFile( const QString& outputFile );
+	void setOutputFile(const QString& outputFile);
 
+	void startDetail(const unsigned int index) { m_detailTimer[index].reset(); }
+	void finishDetail(const unsigned int index) { m_detailTime[index] = m_detailTimer[index].elapsed(); }
+
+	int detailLoad(const unsigned int index) const { return m_detailLoad[index]; }
 
 private:
 	MicroTimer m_periodTimer;
-	int m_cpuLoad;
+	float m_cpuLoad;
 	QFile m_outputFile;
 
+	static const int s_detailCount = 4;                     // set to the actual number of used probes in Mixer.cpp
+	std::array<MicroTimer, s_detailCount> m_detailTimer;    // use arrays to avoid dynamic allocations in realtime code
+#ifdef __GNUC__
```

Comment:
Maybe we better to use this workaround on newer GCC  too? I don't find it that harmful and having the same code may have negative consequences on the long term when compiling on different platforms (e.g. VC or clang)

Reply:
I don't quite understand what are trying to say in the second sentence; do you suggest to keep the workaround even after update to a newer GCC version in our CI (to preserve compatibility with order compilers over long term), or to use the same workaround code for all compilers, not just GCC (since, assuming it works, it would make the code simpler)?

Reply:
I suggest to use the same code everywhere and avoid code duplication. In this case, I would suggest to use this workaround even if the compiler is not gcc. 
Consider the case where one will only update for gcc and then other compilers will break.

Reply:
OK, makes sense. I will keep a comment that explains the workaround, though. Otherwise someone could one day come around and "make it prettier" by removing the extra braces. :)